# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/RenderTextureHostWrapper.cpp
# Commit: 81601a02d9b7
# Full Hash: 81601a02d9b7e658f37f9525cce902eed94d678f
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2020-11-05 16:09:03
# Regressor Bug: 1673983
# File Overlap Count: 2
# Description:
#   Bug 1673983 - Support using D3D11 textures as external compositor surfaces with D3D11 compositor. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D95800
# ==============================================================================

diff -r 72d4bf5b093e -r 81601a02d9b7 gfx/webrender_bindings/RenderTextureHostWrapper.cpp
--- a/gfx/webrender_bindings/RenderTextureHostWrapper.cpp	Thu Nov 05 09:15:05 2020 +0000
+++ b/gfx/webrender_bindings/RenderTextureHostWrapper.cpp	Thu Nov 05 09:15:28 2020 +0000
@@ -72,6 +72,15 @@
   return mTextureHost->AsRenderDXGITextureHost();
 }
 
+RenderDXGIYCbCrTextureHost*
+RenderTextureHostWrapper::AsRenderDXGIYCbCrTextureHost() {
+  EnsureTextureHost();
+  if (!mTextureHost) {
+    return nullptr;
+  }
+  return mTextureHost->AsRenderDXGIYCbCrTextureHost();
+}
+
 RenderTextureHostSWGL* RenderTextureHostWrapper::EnsureRenderTextureHostSWGL()
     const {
   EnsureTextureHost();